src/iosource/pcap/Source.cc

```diff
@@ -197,29 +196,47 @@ bool PcapSource::ExtractNextPacket(Packet* pkt)
 	if ( ! pd )
 		return false;
 
-	const u_char* data = pcap_next(pd, &current_hdr);
+	const u_char* data;
+	pcap_pkthdr* header;
 
-	if ( ! data )
-		{
-		// Source has gone dry.  If it's a network interface, this just means
-		// it's timed out. If it's a file, though, then the file has been
-		// exhausted.
-		if ( ! props.is_live )
-			Close();
+	int res = pcap_next_ex(pd, &header, &data);
 
+	switch ( res ) {
+	case -2:
```

Comment:
Use symbolic names? "man pcap" calls these `PCAP_ERROR_BREAK` and ` PCAP_ERROR`

Reply:
My `man pcap` and `man pcap_next_ex` on a couple systems do say `-2` and `-1` directly.  Docs seems to have changed in 2018, but those definitions do exist since libpcap 1.0.0, so sure, changed it.